src/components/data-management/ProjectDetails.jsx

```diff
@@ -342,6 +333,35 @@ const ProjectDetails = ({ width, height }) => {
     dispatch(deleteMetadataTrack(name, activeProjectUuid));
   };
 
+  const createUpdateObject = (value, metadataKey) => {
+    const updateObject = metadataKey === 'species' ? { species: value } : { metadata: { [metadataKey]: value } };
+    return updateObject;
+  };
+
+  const setCells = (value, metadataKey, actionType) => {
+    if (!MASS_EDIT_ACTIONS.includes(actionType)) return;
+    const updateObject = createUpdateObject(value, metadataKey);
+
+    const canUpdateCell = (sampleUuid, action) => {
+      if (action !== 'REPLACE_EMPTY') return true;
+
+      const isSpeciesEmpty = (uuid) => metadataKey === 'species' && !samples[uuid].species;
+      const isMetadataEmpty = (uuid) => metadataKey !== 'species'
+        && (!samples[uuid].metadata[metadataKey]
+          || samples[uuid].metadata[metadataKey] === defaultNA);
```

Comment:
my editor says that this `defaultNA ` is not defined

Reply:
Changed to `DEFAULT_NA`. Updated accordingly